src/plugins/algolia.js

```diff
@@ -0,0 +1,25 @@
+const ALGOLIA_CSS_URL = 'https://cdn.jsdelivr.net/npm/docsearch.js@2/dist/cdn/docsearch.min.css';
+const ALGOLIA_JS_URL = 'https://cdn.jsdelivr.net/npm/docsearch.js@2/dist/cdn/docsearch.min.js';
+const ALGOLIA_INPUT_SELECTOR = '#algolia-search-input';
+
+function buildAlgoliaInitScript(apiKey, indexName) {
+  return `<script>
+    function initAlgolia() {
+      docsearch({
```

Comment:
Should we also allow the user to specify the [`debug` key](https://community.algolia.com/docsearch/styling.html#debugging)? It may be useful for debugging CSS styles on the Agolia search menu.

More generally, should we expose all the options given in the [`docsearch.js` API](https://community.algolia.com/docsearch/behavior.html)? I can see [`agoliaOptions`](https://community.algolia.com/docsearch/behavior.html#algoliaoptions) being useful eg. to control the number of results returned per search, but the other options are function hooks which I am unsure if our current plugin system can handle, since `site.json` can't take functions.

Reply:
That's a good point.

I'm not sure how useful the `debug` property will be, since we aren't really providing users a way to set up custom CSS for the dropdown anyway.

Reply:
> I'm not sure how useful the `debug` property will be, since we aren't really providing users a way to set up custom CSS for the dropdown anyway.

@marvinchin the CSS for the dropdown can be overridden by specifying the relavant [`.agolia-autocomplete`](https://community.algolia.com/docsearch/styling.html) styles in a layout's `styles.css`.

Reply:
Thanks for pointing that out, I must've missed that part.

Reply:
I've added support for the additional options.